### DISCUSSION TITLE
Term with ID not Found

### DISCUSSION THREAD

**Discussion by nthoman** [microsoft-search/pnp-modern-search#2403] [General]
![image](https://user-images.githubusercontent.com/48287847/193313075-af4c1475-a9fb-49b7-bab8-caf9bf646392.png)

I merged some terms in my term store in SharePoint Admin Center and now the Search Filter result list is still showing up terms that are not available anymore. I have confirmed the documents no longer have the old term mapped and I re-indexed the site but it's still showing up. Can we just have a switch to turn off NOT FOUND terms in the filter config?

**Comment by wobba** [comment]
@nthoman remember that re-indexing is a queue operation, not an instant operation. It can take as much as 2-3 days before being reflected in search. If you use the `Get-PnPSearchCrawlLog` cmdlet you can see when the item was last indexed - assuming you have the rights to view the crawl log.

**Reply by nthoman** [reply]
Yep... it's been a few months since I first noticed this issue and I may still have some documents tagged with the old tag. However... since we know indexing can be delayed, should just give an option to not show the error all together to improve the end user experience.

**Reply by wobba** [reply]
It's not an error per say, and do you have a suggestion on how the web part should best show this? I imagine the text could come back in any language, so matching on the text seems tricky to make it robust.

**Reply by nthoman** [reply]
Maybe there is a generic ID code or something we can match on. I know when I look at the document library directly and filter there on keywords, I don't see this "error".

![image](https://user-images.githubusercontent.com/48287847/194883846-cf411aab-ac4f-4878-a32b-cd9d9809eb2a.png)

**Reply by wobba** [reply]
Library calls directly to term store API's. Search rely on the value coming from the search index. If there is a way to query on the old id and get the new term, maybe we can change? I'm unaware of the underlying mechanisms in the term store when terms merge.

**Reply by nthoman** [reply]
Is there a way hunt down where this missing Term is located? It's been a few months now and I still can't get the merged terms to clear out. No matter how much I re-index. So do you think this is a flaw in the Term Store API? Again... only PnP search has this problem. When you look at the filter options directly on the document library the merged term ID's do not show up there.

**Reply by wobba** [reply]
It's not only PnP, it's search :) If you edit/update each item where this happens I think it might resolve itself. Or maybe edit the hidden taxonomy list in the site to reflect the right value an item points to, and then reindex.

**Reply by nthoman** [reply]
Focusing on this this week..... doing some more digging. For this document library I am also using Syntex to classify and tag the documents... so am wondering if that might be introducing this "Missing Tag". I am going to retrain the Syntex model and then re-classify the documents to see if that cleans it up.

**Reply by ileana2019** [reply]
I have the same error: "Term with ID not found". I have the site term sets and the refinable strings provisioned with PnP PowerShell and site template (Invoke-PnPSiteTemplate). @nthoman do you have a solution for this? Thank you!

**Reply by nthoman** [reply]
So to fix this I had to force all documents that had the old tag to the corrected Tag. I ended up creating a Power Automate Workflow to do this. I also was using Syntex Classify and Extract to do some of the document tagging. So I had to re-classify them. Do this I built a PowerShell script that would leverage PnP SharePoint Search and then kick of Syntex to re-classify the documents. 

"772cb1ce-7bd7-4f03-b39d-2f0eb14f7c14" was one of my screwed up Tags.

> #Config Variables
> $SiteURL = "https://(Instance).sharepoint.com/sites/(mysite)"
> $SearchQuery = "772cb1ce-7bd7-4f03-b39d-2f0eb14f7c14 Path:" + $SiteURL
> 
> Connect-PnPOnline -Url $SiteURL -UseWebLogin
> 
> Write-Host "Searching for: $SearchQuery" -ForegroundColor Cyan
> 
> 
>     $SearchResults = Submit-PnPSearchQuery -Query $SearchQuery -All
> 
>     $Results = @()
>     foreach($ResultRow in $SearchResults.ResultRows) 
>     { 
>         #Get All Properties from search results
>         $Result = New-Object PSObject 
>         $ResultRow.GetEnumerator()| ForEach-Object { $Result | Add-Member Noteproperty $_.Key $_.Value} 
>         $Results+=$Result
>     }
> 
>     #$Results.OriginalPath
>     $RCount = $Results.Count
>     $RTotal = $Results.Count
> 
>     foreach($File in $Results.OriginalPath)
>     {
>         Write-Host "Submitting: $file" -ForegroundColor Green
>         Write-Host "File: $RCount of $RTotal" -ForegroundColor Green
>         Request-PnPSyntexClassifyAndExtract -FileUrl $File
>         $RCount = $Rcount-1
>     }
